app/models/admin_setting.rb

```diff
@@ -71,8 +71,11 @@ def self.perform(method, *args)
 
   private
 
-  def expire_cached_settings
-    Rails.cache.delete("admin_settings")
+  def recache_settings
+    # Reload the record; if the default skin was just created and has a
+    # closed file handle from generating a preview image, it cannot be
+    # serialized for caching.
+    Rails.cache.write("admin_settings", self.reload)
```

Comment:
Sorry, I guess I should have mentioned that reloading is a little more complex than it seems. You can't reload outside of the transaction, or you might get stale data. But to avoid writing incorrect data in the event that the transaction fails, you probably want to write to the cache outside of the transaction. So you probably either need multiple callbacks (an `after_save` one to load the fresh copy, and an `after_commit` one to actually write it) or a single `after_save` hook that calls `AfterCommitEverywhere.after_commit` to execute a block after the commit has finished.

(It might be simpler to just stick with the comment.)

Reply:
I'll give the 2-callback approach a try, it could be more foolproof than only adjusting the specs and leaving comments.